test/filters/unit_tests-macos.filter

```diff
@@ -4,3 +4,5 @@
 ## When adding a new filter to this list, please include a short description of
 ## why the filter is required and create an associated tracking issue.
 ##
+
+-ProcessMemoryDumpTest.CountResidentBytesInSharedMemory # Doesn't work with Rosetta
```

Comment:
in other filter files we have comments followed by the filters, not on the same line.

a little more info here also would be nice, i.e. mention that we want to build x64 builds on arm64, but this exact test doesn't work.

Reply:
hm, there is a problem. it's from `base_unittests` and we don't launch them directly.
But we added this file because we want to run brave `IsMemoryAllocatorDumpNameInAllowlist` test.
I've moved `IsMemoryAllocatorDumpNameInAllowlist` to a separate file.

Now we don't run other chromium `ProcessMemoryDumpTest` (the same as the other `base_unittests`).